pkg/controller/component/controller.go

```diff
@@ -158,22 +143,47 @@ func (r *ReconcileComponent) Delete(resource controller2.Resource) error {
 	return nil
 }
 
-// Check if the Pod Condition is Type = Ready and Status = True
-func (r *ReconcileComponent) isPodReady(pod *corev1.Pod) bool {
-	for _, c := range pod.Status.Conditions {
-		if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
-			return true
-		}
-	}
-	return false
-}
-
-// Check if the TaskRun Condition is Type = SUCCEEDED and Status = True
-func (r *ReconcileComponent) isBuildSucceed(taskRun *taskRunv1alpha1.TaskRun) bool {
-	for _, c := range taskRun.Status.Conditions {
-		if c.Type == apis.ConditionSucceeded && c.Status == corev1.ConditionTrue {
-			return true
+func (r *ReconcileComponent) SetPrimaryResourceStatus(primary controller2.Resource, statuses []controller2.DependentResourceStatus) (needsUpdate bool) {
```

Comment:
Is it documented what are the component statuses and status about the dependent resources ?

Reply:
Nothing is documented at this point.